Fix animation handling.

GraionDilach opened the pull request:
Revert animations to their Beamdog-allocated IDS names and slots to prevent random slot overwrite:
- 0xE272 BARGHEST -> 0xE280 WORG (seems unused in the mod?)
- 0xE510 KAZUUL -> 0xE279 OROG_SHAMAN
- 0xEC00 SKELETON_GRANDLORD -> 0xE242 ANIMATED_PLATE_GREEN
- 0xE320 GHAST_GREATER -> 0xE241 ANIMATED_PLATE_DARK
- 0xE300 GHOST -> 0xE240 ANIMATED_PLATE_BRONZE
- 0xEE00 ZOMBIE_YELLOW -> 0xE271 GOBLIN_SHAMAN
- 0xE230 BEETLE_RHINOCEROS -> 0xE250 GOLEM_IRON_IWD

Use ANVIL animate.ids prefix, modder prefixes and dynamic allocation for animations not existing in the BG2EE animation array:
- ANVIL_GHOUL_QUEEN
- ANVIL_ANIMATED_PLATE_BRONZE (clones ANIMATED_PLATE_BRONZE with translucency to main the status quo caused by the currently used GHOST slot and it's settings)

Included missing audio for animations:
- 0xE250 GOLEM_IRON_IWD
- 0xE271 GOBLIN_SHAMAN
- 0xE279 OROG_SHAMAN
- 0xE280 WORG
- ANVIL_GHOUL_QUEEN

Also deleted the IC_ICE_GOLEM reference, the EEs did away the IC prefix, the animation is listed as GOLEM_ICE on the platform.

References:
https://www.gibberlings3.net/forums/topic/28835-toss-your-semi-useful-weidu-macros-here/page/2/#comment-254894
https://github.com/GwendolyneFreddy/Infinity-Animations/tree/master/infinityanimations/documentation

GraionDilach committed:
Don't overwrite animations.

Don't overwrite bonebat assets if already included.
Don't overwrite Goblin Worgrider with Worg.
Dynamically allocate an IWD slot for the Ghoul Queen.

Animation folders renamed to match ANIMATE.IDS entries.
Missing audio included for Worg and the Ghoul Queen.

GraionDilach committed:
Reimport Orog Shaman assets from SoD.

Also updated Kazuul for correct slot.

GraionDilach committed:
Reimport Animated Plate, Green assets from SoD.

Also update creature files to stop using Wight, Gray slots.

GraionDilach commented:
@critto-bg  @SparrowJacek 
Is the SKELETON_LORD transparency effect intentional? Because ANIMATED_PLATE_BRONZE lacks and never intended that. It comes from the GHOST animation INI setup.

If it is intended, I'll set up the PR so that it assigns a custom IWD animation slot for the variant.

EDIT: Decided to just keep the status quo on this one.

GraionDilach committed:
Reimport Animated Plate, Bronze assets from SoD.

Also update creature files to stop using Ghost slots statically.
Also added translucent variant of INI to maintain the status quo.

If the translucency is unintended, just revert the during-install
patching and the dynamic slot allocation.

GraionDilach committed:
Reimport Animated Plate, Dark assets from SoD.

Also update creature files to stop using Ghast, Greater slots.

GraionDilach committed:
Reimport Golem, Iron (IWD) assets from IWDEE.

Also update creature files to stop using Beetle, Rhinoceros slots.

SparrowJacek commented:
Thank you very much for your help! I will analyze your changes and hopefully understand how to operate in the future. I don't think that transparency effect was intended, I am not not even sure how it was achieved.

In general I was just happy that my attempts at introducing animations from other IE games succeeded, I never dug deeper to understand how to introduce them properly. And it was all around 7 years ago, so forgot all about it...

GraionDilach committed:
Reimport Goblin Shaman assets from SoD.

Also update creature files to stop using Zombie, Yellow slots.

GraionDilach committed:
Delete misinformation.

ANISND.IDS is no longer used in the EE 2.0+ engine.
IC_ICE_GOLEM is called GOLEM_ICE in the EEs, because they omitted the
IC prefix.

GraionDilach commented:
I'm done with the cleanup. This is ready for review.

GraionDilach commented on ImprovedAnvil/lib/creatures/01_copy_new.tph:
```diff
@@ -191,54 +191,67 @@ COPY ~ImprovedAnvil/resources/creatures/new/S!gcoin.cre~         ~override/S!gco
 COPY ~ImprovedAnvil/resources/creatures/new/S!skel.cre~         ~override/S!skel.cre~
   SAY NAME1     @411
   SAY NAME2     @411
+  WRITE_LONG 0x28 (IDS_OF_SYMBOL (~animate~ ~ANVIL_ANIMATED_PLATE_BRONZE~))
```
If you want to delete the unintended transparency, you can just delete these lines. The creature files are updated to ANIMATED_PLATE_BRONZE statically.

GraionDilach commented on ImprovedAnvil/resources/animations/ini/anvil_animated_plate_bronze.ini:
```diff
@@ -0,0 +1,59 @@
+[monster_icewind]
+detected_by_infravision=0
+resref=MANI
+translucent=1
```
This is the tag copied from Ghost's own INI animation file which achieves the transparency.

GraionDilach commented on ImprovedAnvil/lib/core/animations.tph:
```diff
@@ -0,0 +1,255 @@
+	COPY ~ImprovedAnvil\resources\animations\bonebat~ ~override~
+END
+
+// Ghoul Queen, Acaste (converted from PST to an IWD slot) (Cuv-conversion? Extended Animations?)
```
I'm not sure where this was sourced from. This a guess. Cuv converted PST animations to the IWD format at one point, but his pack kept the native PST scale (which is bigger than the other IE scales.)

IIRC Extended Animations included downsamples to the common Infinity Engine scale.

SparrowJacek replied:
From what I remember, I extracted the original animation BAMs and used some program, BAMresizer, or something, to scale them.

GraionDilach replied:
Huh, I need to doublecheck this then, because the PST bam and frame structure differs from all the other setups and this might be still wrong.

GraionDilach replied:
Doublechecked, this still looks correct for the IWD animation slot.

GraionDilach committed:
Delete skeleton animations being reset manually.

Animation values should be either set statically or during the
copy-to-override step because this makes debugging harder.

critto-bg commented:
Finally found the time to check and review this (thanks to @SparrowJacek for the help). The worg and the goblin shaman animations will be used in the upcoming v7 contents.

Transparency seems to be an unintended side effect that ended up being released as a feature. We'll probably fix it even though nobody seemed to mind it since the change had been introduced.

Once again, thanks for the help.
